File: Data Structures & Algorithms/three-integer-sum/submission-2.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);

        List<List<Integer>> res = new ArrayList<>();
        for(int i=0;i<nums.length;i++){
            if ((i>0 && nums[i] == nums[i-1]) || nums[i] > 0) continue;

            int target = -nums[i];
            int l = i, r = nums.length - 1;
            while (l < r){
                if (nums[l] + nums[r] < target) l++;
                else if (nums[l] + nums[r] > target) r--;
                else {
                    List<Integer> cur = new ArrayList<>(Arrays.asList(nums[l], nums[r], nums[i]));
                    res.add(cur);

                    while (l < nums.length - 1 && nums[l] == nums[l+1]) l++;
                    while (r > 0 && nums[r] == nums[r-1]) r--;

                    l++;
                    r--;
                }
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        
        res, count = -1, 0

        for num in nums:
            if num == res:
                count += 1
            elif count == 0:
                count = 1
                res = num
            else:
                count -= 1
        
        return res
        